### PR TITLE
[fix](hyperscan) Fix hyper scan fall back to re2

### PR DESCRIPTION
**Pull request by zhiqiang-hhhh**
### What problem does this PR solve?

* Core modification
When hyper scan failed, we should not set_error in FunctionContext.
Since set_error will try cancel query, but actually we want to fall back to re2 in this case.

* Some refactor
Rename FunctionRegexp so that we can distinguish regexp match with regexp_extract.

* Reproduce
```cpp
SELECT * FROM regexp_test_chinese WHERE city REGEXP "^上海｜^北京" ORDER BY id;
```
Note, the `|` in above sql is a Chinese character.

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Review by xiaokang — DISMISSED** [review]
LGTM

File: be/src/vec/functions/like.cpp [hunk 1]
```diff
@@ -499,9 +496,6 @@ Status FunctionLikeBase::hs_prepare(FunctionContext* context, const char* expres
         hs_free_database(*database);
         *database = nullptr;
         *scratch = nullptr;
-        if (context) {
```
Comment (cambyzju) [inline be/src/vec/functions/like.cpp:502]: please leave a comment here, avoid somebody add these codes back again next time.

Comment (zhiqiang-hhhh) [reply be/src/vec/functions/like.cpp:502]: OK，I will add some test and comment.

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by cambyzju — APPROVED** [review]

**Pull request merged**